packages/ui-components/src/components/form/Switch.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+  <div class="flex items-center">
+    <HeadlessSwitch
+      v-model="enabled"
+      class="relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary"
+      :class="{ 'bg-primary': enabled, 'bg-primary-muted': !enabled }"
+    >
+      <div class="absolute inset-0 flex items-center gap-2 px-1 text-white">
+        <CheckIcon class="h-5 w-5 drop-shadow-md" />
+        <XMarkIcon class="h-5 w-5 drop-shadow-md" />
+      </div>
+      <span
+        class="scale-95 pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow transform ring-0 transition ease-in-out duration-200"
+        :class="{ 'translate-x-5': enabled, 'translate-x-0': !enabled }"
+      ></span>
+    </HeadlessSwitch>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, watch } from 'vue'
+import { Switch as HeadlessSwitch } from '@headlessui/vue'
+import { CheckIcon, XMarkIcon } from '@heroicons/vue/24/solid'
+
+const props = defineProps<{
```

Comment:
Pretty much the entire script setup block can just be simplified to a single `defineModel<boolean>()` call.

Reply:
Another one to check before I mark as resolved. I replaced the script (except imports) with:
const enabled = defineModel<boolean>()